test/e2e/e2e_test.go

```diff
@@ -92,6 +106,15 @@ func TestE2E(t *testing.T) {
 	}
 
 	gomega.RegisterFailHandler(ginkgo.Fail)
+
+	// Ensure all pods are running and ready before starting tests (otherwise,
+	// cluster infrastructure pods that are being pulled or started can block
+	// test pods from running, and tests that ensure all pods are running and
+	// ready will fail).
+	if err := waitForPodsRunningReady(api.NamespaceDefault, minStartupPods, podStartupTimeout); err != nil {
```

Comment:
We're waiting for the system pods to be created, pulled and running, but for paranoia you don't want to exit the check way too early and exit when there have been 0 pods created. In practice, `minStartupPods` closes a highly theoretical race, at the expense of increasing confusion.